.github/workflows/buf-ci.yaml

```diff
@@ -0,0 +1,23 @@
+name: Buf CI
+on:
+  push:
+    paths:
+      - 'proto/**'
```

Comment:
Why don't we use exactly the same patterns as what we have in the documentation just to cover our bases? push:
  paths:
    - '**.proto'
    - '**/buf.yaml'
    - '**/buf.lock'
    - '**/buf.md'
    - '**/README.md'
    - '**/LICENSE'

Reply:
Those are a catch all. I can add a `buf.lock` but `**.proto` would trigger on test files unrelated to this module.

Reply:
LICENSE and REAME.md were removed as they don't affect the module. Only the `proto/buf.md` `proto/LICENSE` which are covered by `proto/**`

Reply:
That's not true (or shouldn't be) - there is no `proto/buf.md`, and `proto/LICENSE` is just a symlink to `LICENSE (and not sure how that behaves). `README.md` is also valid now, and given that we're on `v2` of `buf.yaml`, the `README.md` should be read directly from root (so should the `LICENSE`, and the symlink shouldn't be needed anymore). If the symlink is needed (ie the `LICENSE` disappears from the module on push), then that's a bug with `buf`.

We also need to have our standard recommendation for customers on what to put here, and we should be doing exactly that.

Reply:
Took it as intended behaviour as https://buf.build/bufbuild/buf does not have README content. Will double check symlink behaviour.

Reply:
Ed, can you remind me why we decided to add these filters at all? Our default recommendation in the GHA is to push all commits, not just those that touch proto files, so that would be an argument to not have any filters here. If we decide that we should have the filters here, that would beg the question if we should update our default recommendation for the GitHub action

Reply:
> README.md is also valid now, and given that we're on v2 of buf.yaml, the README.md should be read directly from root (so should the LICENSE, and the symlink shouldn't be needed anymore).

They are read from the module root bucket, not the workspace root bucket:

https://github.com/bufbuild/buf/blob/abd329df673d23cb72a4332598d3d70cf5ae7fbb/private/buf/bufworkspace/workspace_targeting.go#L560-L564

I filed an issue about this in regards to this conversation https://github.com/bufbuild/buf/issues/3112

Reply:
Added back README and LICENSE. Will work as intended when the above issue is fixed.

Reply:
Ok so there are two differences from default recommended behavior of GHA
1. We have exclude paths in the first place
2. We are using non-default exclude paths because there are nested test proto files that we don't care about.

Regarding #1: we could eliminate the exclude paths and just push for every change (nothing incorrect about that).

Regarding #2: Similarly there is nothing incorrect about pushing changes if unrelated proto files change. 

@bufdev do you have a preference between having no exclude paths at all vs using exclude paths we document?